### PR TITLE
NEXUS-42970 add examples and doc for GCP

### PR DESCRIPTION
**Pull request by vfedoriv**
Jira: https://sonatype.atlassian.net/browse/NEXUS-42970

Note: based on https://github.com/sonatype/nxrm3-ha-repository/pull/66 commit (first commit in PR)

### CODE REVIEW

File: nxrm-ha/README.md [hunk 1]
```diff
@@ -320,6 +321,244 @@ Azure Key Vault is disabled by default. If you would like to store your database
...
+          enabled: true
+          ...
+          defaultRule: true
+          additionalRules: null
+          ingressClassName: gce
+          ...
+          annotations:
+            kubernetes.io/ingress.class: "gce"
+        ```
+   4. Enable Nexus NodePort service
```
Comment (bobotimi) [inline nxrm-ha/README.md:360]: Is it possible to use ClusterIP as well on GKE? If so consider changing to 'Enable Nexus NodePort/ClusterIP service'

File: nxrm-ha/README.md [hunk 2]
```diff
@@ -320,6 +321,244 @@ Azure Key Vault is disabled by default. If you would like to store your database
...
+            adminPassword: admin123 #You should change this when you login for the first time
+        ```
+
+   8. Set license (use base64 to encode file)  
+        ```
+          license:
+            name: test-users.lic
+            licenseSecret:
+              enabled: true
+              fileContentsBase64: cylwwtYx6Fg1CUa9yGqBuhGhgc4IS67Ha/+uvxSpA  == your base64 encoded license file == Gd7Z3+WS/0LIeugxSIa+ZDqtg7AR+U3d9ZJA==
```
Comment (bobotimi) [inline nxrm-ha/README.md:409]: You can also specify the license file with your helm command as in:

--`set-file secret.license.licenseSecret.file=<path to your license file>`

File: nxrm-ha/README.md [hunk 3]
```diff
@@ -320,6 +321,244 @@ Azure Key Vault is disabled by default. If you would like to store your database
...
+          --role roles/iam.workloadIdentityUser \
+          --member "serviceAccount:<your-project-id>.svc.id.goog[<your-namespace>/<your-k8s-service-account>]"
+        ```
+   * 2.f
+        Annotate the Kubernetes Service Account:
+    
+        ```
+            kubectl annotate serviceaccount <your-k8s-service-account> \
+            --namespace <your-namespace> \
+            iam.gke.io/gcp-service-account=<your-gsa-name>@<your-project-id>.iam.gserviceaccount.com
```
Comment (bobotimi) [inline nxrm-ha/README.md:508]: In both AWS and Azure usages of the helm chart, it is possible to annotate the service account by including it in the values.yaml i.e. serviceAccount.annotations. Does this not work for GKE?

File: nxrm-ha/README.md [hunk 4]
```diff
@@ -320,6 +321,244 @@ Azure Key Vault is disabled by default. If you would like to store your database
...
+    
+         ```
+         service:  #Nexus Repo NodePort Service
+           annotations: {}
+           nexus:
+             enabled: true
+         ```
+
+   5. Now you can update secrets separately (follow steps 6, 7, 8 below) 
+      OR install ESO ([External Secret Operator]https://external-secrets.io/latest/introduction/getting-started/)
```
Comment (bobotimi) [inline nxrm-ha/README.md:370]: We should include a text that says 'ESO is the recommended approach for production'

**Review by bobotimi — APPROVED** [review]
Nice work. Just a few comments.

File: nxrm-ha/README.md (hunk 1 above)
Comment (vfedoriv) [reply nxrm-ha/README.md:360]: yes, updated

File: nxrm-ha/README.md (hunk 2 above)
Comment (vfedoriv) [reply nxrm-ha/README.md:409]: added comment in doc

File: sample-gcp-ha-yamls/gcp-ha-docker-service.yaml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: nxrm-nexus-docker-service
+  namespace: nxrm
+  labels:
+    app: nxrm
+spec:
+  type: NodePort
```
[the remark this answers is not in the document]
Comment (vfedoriv) [reply sample-gcp-ha-yamls/gcp-ha-docker-service.yaml:30]: updated

File: nxrm-ha/README.md (hunk 3 above)
Comment (vfedoriv) [reply nxrm-ha/README.md:508]: yes, it's possible and it's described in the doc above.
Updated - I marked this step as optional and added a link to a section where we update service account in values.yaml

File: nxrm-ha/README.md (hunk 4 above)
Comment (vfedoriv) [reply nxrm-ha/README.md:370]: Yep, updated

**Review by VladimirGrab — APPROVED** [review]

**Pull request merged**